src/mds/Server.cc

```diff
@@ -5546,6 +5546,19 @@ void Server::handle_client_setdirlayout(MDRequestRef& mdr)
 }
 
 // XATTRS
+int Server::check_if_pool_is_empty(const OSDMap& osdmap, int64_t pool_id)
+{
+  /* Since we don't allow to create pool level snaps for all pools associated
+   * with a CephFS, we also block a pool with pre-existing snaps to be attached
+   * to a CephFS.
+   */
+  if (const auto *p = osdmap.get_pg_pool(pool_id); p->snaps.size() > 0) {
+    dout(20) << "pool(" << pool_id << ") already has snaps; can't attach to fs" << dendl;
```

Comment:
should this be derr ?

Reply:
this is not a fatal error, so its not a good idea to tag it as `ERR` in the logs
in fact we are avoiding errors by blocking the addition of such pools to the fs